Duplicati/Library/Main/Operation/RestoreHandler.cs

```diff
@@ -720,7 +720,7 @@ private static void ScanForExistingSourceBlocks(LocalRestoreDatabase database, O
                                                 using (var sourcefile = m_systemIO.FileOpenRead(source.Path))
                                                 {
                                                     sourcefile.Position = source.Offset;
-                                                    var size = sourcefile.Read(blockbuffer, 0, blockbuffer.Length);
+                                                    int size = Library.Utility.Utility.ForceStreamRead(sourcefile, blockbuffer, blockbuffer.Length);
                                                     if (size == targetblock.Size)

```

Comment:
This condition feels odd to me.  Is `sourcefile` guaranteed to be `block.Size` bytes in length?

Reply:
I think this is the same logic as before. We look for a specific block, and ignore those that do not match.